config/routes.rb

```diff
@@ -20,6 +20,7 @@
   get "/register", to: "registration#index", as: :registration
   get "/register/auth", to: "registration#authenticate", as: :registration_auth
   post "/user/registered_email", to: "spree/users#registered_email"
+  get "/user/set_lang/:lang", to: "application#set_lang", :as => :set_lang
```

Comment:
The ApplicationController tends to become too complex. Adding this method here means that every controller inherits this method even though no other controller needs it. I would suggest to create a new controller for this, the `LocalesController`. The locale can be stored in the database (user record) but it's also stored in a cookie which is important for guest users. So I would say that putting it under `/user/..` is misleading.

```suggestion
  resources :locales, only: [:show]
```

It's still not 100% in line with Rails conventions because `get` usually doesn't change anything but it's the easiest way of doing it. Rails does support POST in links but it relies on Javascript: https://stackoverflow.com/a/4752019/3377535
I would still go with the normal get request and therefore I used `show` in the example above.
It's like, "show me [all pages in] Spanish".

Reply:
Makes a lot of sense. Committed: aa982fe8d